packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -81,16 +81,27 @@ export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
       };
     }, [isOpen]);
 
-    // Prevents document from scrolling when the dialog is open.
     const originalBodyOverflow = React.useRef('');
+    React.useEffect(() => {
+      if (isOpen) {
+        originalBodyOverflow.current = document.body.style.overflow;
+      }
+    }, [isOpen]);
+
+    // Prevents document from scrolling when the dialog is open.
     React.useEffect(() => {
       const ownerDocument = dialogRef.current?.ownerDocument;
-      if (!ownerDocument || !preventDocumentScroll) {
+      // If there is no `ownerDocument` or `preventDocumentScroll` is false or
+      // document body originally has `overflow: hidden` (possibly from other/parent dialog), then do nothing.
+      if (
+        !ownerDocument ||
+        !preventDocumentScroll ||
+        originalBodyOverflow.current === 'hidden'
```

Comment:
Can you elaborate how/why this fixes the bug (*"When only main modal is open, background behind is scrollable"*)? It's not really clear from the description.

Reply:
Yeah, I couldn't reproduce this part when having modal inside modal, maybe it is something specific to their use case.

Reply:
then what does your PR fix? i'm confused

Reply:
It fixes: when you open parent modal then child modal, close child modal and parent modal, then body won't be scrollable since child modal useEffect cleanup function resets body overflow to hidden.